[BUG] 71.0.0 fails with `backports.tarfile` on Python 3.8

### setuptools version

setuptools==71.0.0

### Python version

Python 3.8

### OS

Linux

### Additional environment information

_No response_

### Description

CI jobs started tripping when using setuptools 71, released two hours ago.
- https://github.com/crate/cratedb-toolkit/pull/201

### Expected behavior

CI builds work as expected.

### How to Reproduce

No reproducer yet, but a stacktrace on GHA.

### Output

```console
ImportError: cannot import name 'tarfile' from 'backports' 
  (/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/backports/__init__.py)
```
-- https://github.com/crate/cratedb-toolkit/actions/runs/9982521975/job/27589490156?pr=196#step:6:229

I'm thinking something like:

```diff
diff --git a/setuptools/__init__.py b/setuptools/__init__.py
index 8b0e494f0..fbc41f205 100644
--- a/setuptools/__init__.py
+++ b/setuptools/__init__.py
@@ -7,6 +7,8 @@ import sys
 from typing import TYPE_CHECKING
 
 sys.path.extend(((vendor_path := os.path.join(os.path.dirname(os.path.dirname(__file__)), 'setuptools', '_vendor')) not in sys.path) * [vendor_path])  # fmt: skip
+# workaround for #4476
+sys.modules.pop('backports', None)  # noqa: E402
 
 import _distutils_hack.override  # noqa: F401
 import distutils.core
```

Would do the trick. Unfortunately, that introduces 10 new nitpicky E402 lint errors:

```
 setuptools main 🐚 ruff check --output-format concise
setuptools/__init__.py:13:1: E402 Module level import not at top of file
setuptools/__init__.py:14:1: E402 Module level import not at top of file
setuptools/__init__.py:15:1: E402 Module level import not at top of file
setuptools/__init__.py:17:1: E402 Module level import not at top of file
setuptools/__init__.py:18:1: E402 Module level import not at top of file
setuptools/__init__.py:19:1: E402 Module level import not at top of file
setuptools/__init__.py:20:1: E402 Module level import not at top of file
setuptools/__init__.py:21:1: E402 Module level import not at top of file
setuptools/__init__.py:22:1: E402 Module level import not at top of file
setuptools/__init__.py:23:1: E402 Module level import not at top of file
Found 10 errors.
```

--- a/newsfragments/4476.bugfix.rst
+++ b/newsfragments/4476.bugfix.rst
@@ -1,0 +1,1 @@
+Reset the backports module when enabling vendored packages.

--- a/pkg_resources/__init__.py
+++ b/pkg_resources/__init__.py
@@ -75,6 +75,8 @@ from pkgutil import get_importer
 import _imp
 
 sys.path.extend(((vendor_path := os.path.join(os.path.dirname(os.path.dirname(__file__)), 'setuptools', '_vendor')) not in sys.path) * [vendor_path])  # fmt: skip
+# workaround for #4476
+sys.modules.pop('backports', None)
 
 # capture these to bypass sandboxing
 from os import utime

--- a/ruff.toml
+++ b/ruff.toml
@@ -43,6 +43,11 @@ ignore = [
 	"ISC002",
 ]
 
+# Suppress nuisance warnings about E402 due to workaround for #4476
+[lint.per-file-ignores]
+"setuptools/__init__.py" = ["E402"]
+"pkg_resources/__init__.py" = ["E402"]
+
 [format]
 # Enable preview to get hugged parenthesis unwrapping
 preview = true

--- a/setuptools/__init__.py
+++ b/setuptools/__init__.py
@@ -7,6 +7,8 @@ import sys
 from typing import TYPE_CHECKING
 
 sys.path.extend(((vendor_path := os.path.join(os.path.dirname(os.path.dirname(__file__)), 'setuptools', '_vendor')) not in sys.path) * [vendor_path])  # fmt: skip
+# workaround for #4476
+sys.modules.pop('backports', None)
 
 import _distutils_hack.override  # noqa: F401
 import distutils.core

